tasks/shared_library_posting_to_queue/shared_lib_posting_to_status_queue.py

```diff
@@ -0,0 +1,98 @@
+from enum import Enum
+import json
+import boto3
+from typing import Dict,Any, Optional
+import time
+
+class RequestMethod(Enum):
+    POST = 'post'
+    PUT = 'put'
+
+class ORCA_STATUS(Enum):
+    """
+    Defines the status of ORCA DB?
+    """
+    PENDING = 1
+    STAGED = 2
+    SUCCESS = 3
+    FAILED = 4
+
+def post_status_for_job_to_queue(job_id: str, granule_id: str, status_id: ORCA_STATUS,
+                                 request_time: Optional[str], completion_time: Optional[str],
+                                 archive_destination: Optional[str],
+                                 request_method: RequestMethod, db_queue_url: str,
+                                 max_retries: int, retry_sleep_secs: float):
+    new_data = {'job_id': job_id, 'granule_id': granule_id}
+    if status_id is not None:
+        new_data['status_id'] = status_id
+    if request_time is not None:
+        new_data['request_time'] = request_time
+    if completion_time is not None:
+        new_data['completion_time'] = completion_time
+    if archive_destination is not None:
+        new_data['archive_destination'] = archive_destination
+
+    post_entry_to_queue('orca_recoveryjob',
+                        new_data,
+                        request_method, db_queue_url, max_retries, retry_sleep_secs)
+
+def post_status_for_file_to_queue(job_id: str, granule_id: str, filename: str, key_path: Optional[str],
+                                  restore_destination: Optional[str],
+                                  status_id: ORCA_STATUS, error_message: Optional[str],
+                                  request_time: Optional[str], last_update: str,
+                                  completion_time: Optional[str],
+                                  request_method: RequestMethod,
+                                  db_queue_url: str,
+                                  max_retries: int, retry_sleep_secs: float):
+    new_data = {'job_id': job_id,
```

Comment:
We may want to break this up into required and optional values and construct the dictionary that way as a thought. It would be good to have well defined JSON schemas here of what we expect for input.

```
## Note I am assuming **kwargs as an input to the function here
data = {}
required_keys = ['job_id', 'granule_id', 'filename', status_id, last_update] # Based on required not null fields in database
optional_keys = [completion_time] # Based on nullable fields in database

# Required variables
for key in required_keys:
    if key in kwargs:
        if kwargs[key] is not None:
            data[key] = kwargs[key]
        else:
             # error value cannot be None
    else:
        # Error does not exist and is required
...

# For optional maybe set to None so it is a null field when it comes across
```
 
Also may want to check with @AndrewDorn  on what the input schema may look like, are we going to pass files here to? See comments on previous PR. 
- https://github.com/nasa/cumulus-orca/pull/68#discussion_r605953689
- https://github.com/nasa/cumulus-orca/pull/68#discussion_r606297406

Reply:
Yeah, I haven't gotten a chance to work on that. I can switch to https://bugs.earthdata.nasa.gov/browse/ORCA-166 if that's a blocker.